### PR TITLE
Add spotbugs support

### PR DESCRIPTION
**Pull request by Hydragyrum**
Here's my first stab at adding SpotBugs support, as per #178. Basically copied the FindBugs extension.

I am unsure of how to actually test this on a real project though. Any help on that front would be welcome.

One important note though is that the SpotBugs plugin requires gradle 5.0+, which would imply that the code-quality-tools plugin will also require gradle 5.0+.

### CODE REVIEW

File: src/main/kotlin/com/vanniktech/code/quality/tools/SpotbugsExtension.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+   */
+  var ignoreFailures: Boolean? = null
+
+  /** @since 0.4.0 */
+  var source: String = "src"
+
+  /** @since 0.4.0 */
+  var effort: String = "max"
+
+  /** @since 0.4.0 */
```
Comment (vanniktech) [inline src/main/kotlin/com/vanniktech/code/quality/tools/SpotbugsExtension.kt]: They all need to be updated

Comment (Hydragyrum) [reply src/main/kotlin/com/vanniktech/code/quality/tools/SpotbugsExtension.kt]: From what I can see, SpotBugs uses the same values as FindBugs

Comment (vanniktech) [reply src/main/kotlin/com/vanniktech/code/quality/tools/SpotbugsExtension.kt]: Oh yes but the extension wasn't there in version 0.4.0 hence it should be updated to 0.18.0 (I think that's the next version)

Comment (Hydragyrum) [reply src/main/kotlin/com/vanniktech/code/quality/tools/SpotbugsExtension.kt]: Fair point, will fix

**Comment by Hydragyrum** [comment]
Ok this might be a bit more complicated than anticipated. SpotBugs uses sourceSets to determine what to run, and will crash if we leave these empty.

**Comment by vanniktech** [comment]
Keep me posted!

**Comment by Hydragyrum** [comment]
Sorry for the radio silence, work is being tricky to juggle atm

**Comment by vanniktech** [comment]
Same here. No rush!

**Comment by vanniktech** [comment]
Closing this due to inactivity. Feel free to pick this up again if you're still interested.

**Pull request closed without merging**